### PR TITLE
Add required dependency between Performance and Sessions

### PR DESCRIPTION
**Pull request by leotianlizhan**
This PR is only the first step, as it only registers the dependency so we can use Sessions SDK and session ID is flowing to us, but not yet being used. This PR is based on https://github.com/firebase/firebase-ios-sdk/pull/10682.

- [x] Step 1: register dependency (this PR). 
- [ ] Step 2: make Fireperf's `SessionManager` use the session IDs we now get from SessionsSDK)

### CODE REVIEW

File: FirebasePerformance/Sources/FPRClient.m [hunk 1]
```diff
@@ -340,4 +357,18 @@ - (void)disableInstrumentation {
...
+- (void)onSessionChanged:(FIRSessionDetails *_Nonnull)session {
+  FPRLogDebug(kFPRSessionId, @"Session ID changed: %@", session.sessionId);
+}
+
+- (BOOL)isDataCollectionEnabled {
+  return self.configuration.isDataCollectionEnabled;
+}
+
+- (FIRSessionsSubscriberName)sessionsSubscriberName {
+  return FIRSessionsSubscriberNameCrashlytics;
```
Comment (samedson) [inline FirebasePerformance/Sources/FPRClient.m:371]: Must be Performance

Comment (samedson) [reply FirebasePerformance/Sources/FPRClient.m:371]: I wanted a way to ensure this and `addDependencyWithName:` used the same name, but couldn't think of any outside defining a constant and use it both places

Comment (leotianlizhan) [reply FirebasePerformance/Sources/FPRClient.m]: Good catch, I saw "Registering Sessions SDK subscriber with name: Crashlytics, data collection enabled: true" in the logs but then I thought it's normal, Crashlytics is already integrated, turns out it's just my typo

**Comment by samedson** [comment]
You can change this to merge into sessions-main now: https://github.com/firebase/firebase-ios-sdk/commits/sessions-main

**Review by visumickey — COMMENTED** [review]
The changes look good to me (and exciting). This is great.

Would there be any tests that needs to be done to ensure that the dependencies are met?

File: FirebasePerformance.podspec [hunk 2]
```diff
@@ -62,6 +62,7 @@ Firebase Performance library to measure performance of Mobile and Web Apps.
   s.dependency 'FirebaseCore', '~> 10.0'
   s.dependency 'FirebaseInstallations', '~> 10.0'
   s.dependency 'FirebaseRemoteConfig', '~> 10.0'
+  s.dependency 'FirebaseSessions', '~> 10.0'
```
Comment (samedson) [inline FirebasePerformance.podspec:65]: We should probably actually add a changelog entry for this change. You can write a similar message to https://github.com/firebase/firebase-ios-sdk/pull/10682/files

**Comment by samedson** [comment]
Said this in the other PR:

Screenshot 2023-01-18 at 3 46 33 PM

File: FirebasePerformance.podspec (hunk 2 above)
Comment (leotianlizhan) [reply FirebasePerformance.podspec:65]: Done

**Review by samedson — APPROVED** [review]
LGTM!

**Pull request merged**